src/core/thread/thread_netif.cpp

```diff
@@ -128,6 +130,7 @@ ThreadError ThreadNetif::Up(void)
 #if OPENTHREAD_ENABLE_DNS_CLIENT
         mDnsClient.Start();
 #endif
+        mChildSupervisor.Start();
```

Comment:
We can consider this as a possible future enhancement/optimization. Will require to add hooks to supervisor whenever a child is added and removed. Also a child may switch multiple times between sleepy to non-sleepy after attach. Downside is that it adds to complexity of the code to handle all above cases.